notebooks/sycamore-tutorial-intermediate-etl.ipynb

```diff
@@ -0,0 +1,410 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sycamore.transforms.partition import ArynPartitioner\n",
+    "# Set your Aryn API key. See https://sycamore.readthedocs.io/en/stable/aryn_cloud/accessing_the_partitioning_service.html#using-sycamore-s-partition-transform\n",
+    "\n",
+    "partitioned_docset = (\n",
+    "        initial_docset.partition(partitioner=ArynPartitioner(extract_images=False,  extract_table_structure=True, use_ocr=True))\n",
+    "        .materialize(path=\"./pc-tutorial/partitioned\", source_mode=sycamore.materialize_config.MaterializeSourceMode.IF_PRESENT)\n",
+    ")\n",
+    "partitioned_docset.execute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We make a remote call to APS for each document. APS will take a few minutes to partition these documents, as there are a few hundred pages across the PDF doucments. \n",
+    "\n",
+    "Notice that we include a `materialize` operation and force execution with `execute`. `materialize` writes the DocSet out to disk, and it is configured to be used as a checkpoint in the pipeline. Subsequent operations will read from this materialized DocSet. In this notebook, we will use commands like `show_pages` to visualize the DocSet for educational purposes (a production pipeline wouldn't do this), and these commands force recomputation of the DocSet. If present, Sycamore will use the materialized DocSet instead of recomputing it (and avoiding running the APS step again)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can visualize the pages of the PDF using the `show_pages` function, so you can see how APS chunked each page into elements.\n",
+    "\n",
+    "Note that this requires the `poppler` library to be installed on your system. You can install it on a Mac with `brew install poppler`, or in Debian with `apt-get install poppler-utils`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sycamore.utils.pdf_utils import show_pages\n",
+    "\n",
+    "show_pages(partitioned_docset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Cleaning data\n",
+    "\n",
+    "We often find that there's a little excess whitespace in the textual representation of documents, which can be bad for our subsequent LLM transforms and embeddings. So, we apply a regex on the text representation of each element to coalesce it with the `regex_replace` transform."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sycamore.transforms.regex_replace import COALESCE_WHITESPACE\n",
+    "\n",
+    "regex_docset = partitioned_docset.regex_replace(COALESCE_WHITESPACE)"
```

Comment:
I think it's worth throwing in a `print(COALESCE_WHITESPACE)` in here. People want to see the regex they're applying.

Reply:
Ack